src/main/scala/repository/UserRepository.scala

```diff
@@ -1,11 +1,22 @@
 package repository
 
 import domain.User
+import schema.UserTable
+import slick.jdbc.H2Profile.api._
+import slick.lifted.TableQuery
 
-trait UserRepository[F[_]] extends Repository {
-  def registerUser(username: String): F[User]
+import scala.concurrent.Future
 
-  def getById(id: Long): F[Option[User]]
+class UserRepository(db: Database, users: TableQuery[UserTable]) {
 
-  def getByUsername(username: String): F[Option[User]]
-}
+  def save(user: User): Future[Option[User]] = {
+    db.run(users += user)
+    findByUsername(user.username)
```

Comment:
you can avoid second fetch query, you can create insert query like:
```scala
private val insertQuery = users returning users.map(_.id) into ((user, id) => user.copy(id = id))
...
val action = insertQuery += User(-1, username, address, email)// -1 will be ignored in case of AutoInc sql type
db.run(action)
```

Reply:
also no retries here, you should use slightly modified version of retryAsync which retries some DB SQL exceptions.